test/e2e/apps/job.go

```diff
@@ -718,10 +718,13 @@ done`}
 
 		ginkgo.By("Creating a job")
 		job := e2ejob.NewTestJob("notTerminate", "exceed-active-deadline", v1.RestartPolicyNever, parallelism, completions, &activeDeadlineSeconds, backoffLimit)
+		// set low to TerminationGracePeriodSeconds to ensure quick pod termination
+		// when SIGTERM hits before the trap is registered
+		job.Spec.Template.Spec.TerminationGracePeriodSeconds = ptr.To[int64](1)
 		job, err := e2ejob.CreateJob(ctx, f.ClientSet, f.Namespace.Name, job)
 		framework.ExpectNoError(err, "failed to create job in namespace: %s", f.Namespace.Name)
 		ginkgo.By("Ensuring job past active deadline")
-		err = waitForJobFailure(ctx, f.ClientSet, f.Namespace.Name, job.Name, time.Duration(activeDeadlineSeconds+15)*time.Second, "DeadlineExceeded")
+		err = waitForJobFailure(ctx, f.ClientSet, f.Namespace.Name, job.Name, time.Duration(activeDeadlineSeconds+20)*time.Second, "DeadlineExceeded")
```

Comment:
This was increased to prevent flakiness? Should it have a TODO?

Reply:
Right, I run experiments, here and in https://github.com/kubernetes/kubernetes/pull/125816 by looping the test. I noticed that pod termination may take up to 40s in some cases. I bump it tentatively to 60s.